crates/ruff_python_formatter/src/core/helpers.rs

```diff
@@ -20,9 +20,8 @@ pub fn find_tok(
     locator: &Locator,
     f: impl Fn(rustpython_parser::Tok) -> bool,
 ) -> (Location, Location) {
-    let (source, start_index, end_index) = locator.slice(Range::new(location, end_location));
     for (start, tok, end) in rustpython_parser::lexer::lex_located(
-        &source[start_index..end_index],
+        locator.slice(Range::new(location, end_location)),
```

Comment:
The `Locator` in `ruff_python_ast` has a slightly different API than the version we "forked" into here, hence these changes, e.g., we need to use a slightly different interface for slices.